docroot/modules/custom/mass_admin_pages/src/EventSubscriber/CookieSubscriber.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Drupal\mass_admin_pages\EventSubscriber;
+
+use Drupal\Core\Cache\Context\CacheContextsManager;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\PageCache\RequestPolicyInterface;
+use Drupal\Core\PageCache\ResponsePolicyInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Drupal\Core\EventSubscriber\FinishResponseSubscriber;
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Route;
+
+/**
+ * Response subscriber to add cookies.
+ */
+class CookieSubscriber extends FinishResponseSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(LanguageManagerInterface $language_manager, ConfigFactoryInterface $config_factory, RequestPolicyInterface $request_policy, ResponsePolicyInterface $response_policy, CacheContextsManager $cache_contexts_manager, $http_response_debug_cacheability_headers = FALSE) {
+    parent::__construct($language_manager, $config_factory, $request_policy, $response_policy, $cache_contexts_manager, $http_response_debug_cacheability_headers);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setResponseNotCacheable(Response $response, Request $request) {
+    parent::setResponseNotCacheable($response, $request);
+
+    if (\Drupal::service('router.admin_context')->isAdminRoute()) {
+      $response->headers->clearCookie('ah_app_server', '/');
+
+      // Set cookie to stick file uploads to one server when on edit pages to
+      // address issues with uploading files via the WYSIWYG.
+      // See https://support.acquia.com/hc/en-us/articles/360004147834-Pinning-to-a-web-server-without-using-the-hosts-file#defineacookie
+      if ($request->attributes->get('_route') == 'entity_browser.browse_files') {
```

Comment:
To answer both questions, I did it this way because of the Important disclaimer here: https://docs.acquia.com/cloud-platform/manage/files/broken/#pinning-temporarily-to-a-web-server. It's my understanding as well that this will effectively pin all requests for that session to the same server. So, I wanted to do this with as little footprint as possible and hopefully avoid a future hot patch if for some reason one day one of our web servers starts seeing too much activity and some authors end up stuck on that server, which, of course, we could work around.

Reply:
I think that disclaimer is important only if you are pinning all your requests. We are only pinning editors so no load issues apply IMO.

Reply:
Makes sense. I made the change, and this is ready for review again